crates/ironrdp-web/src/session.rs

```diff
@@ -522,17 +534,38 @@ async fn connect(
     proxy_auth_token: String,
     destination: String,
     pcb: Option<String>,
+    kdc_proxy_url: Option<String>,
 ) -> Result<(connector::ConnectionResult, WebSocketCompat), IronRdpError> {
     let mut framed = ironrdp_futures::SingleThreadedFuturesFramed::new(ws);
 
-    let mut connector = connector::ClientConnector::new(config)
-        .with_server_name(&destination)
-        .with_credssp_network_client(WasmNetworkClientFactory);
-    // .with_static_channel(ironrdp::dvc::Drdynvc::new()); // FIXME: drdynvc is not working
-
-    let upgraded = connect_rdcleanpath(&mut framed, &mut connector, destination, proxy_auth_token, pcb).await?;
-
-    let connection_result = ironrdp_futures::connect_finalize(upgraded, &mut framed, connector).await?;
+    let mut connector = connector::ClientConnector::new(config);
+
+    let (upgraded, server_public_key) =
+        connect_rdcleanpath(&mut framed, &mut connector, destination.clone(), proxy_auth_token, pcb).await?;
+
+    info!("kdc url = {:?}", &kdc_proxy_url);
+
+    let mut wasm_network_client = WasmNetworkClient::new();
+    let network_client: Option<&mut dyn AsyncNetworkClient> = kdc_proxy_url
+        .as_ref()
+        .map(|_| &mut wasm_network_client as &mut dyn AsyncNetworkClient);
+
+    let connection_result = ironrdp_futures::connect_finalize(
+        upgraded,
+        &mut framed,
+        (&destination).into(),
+        server_public_key,
+        network_client,
+        connector,
+        Some(KerberosConfig {
+            kdc_proxy_url: kdc_proxy_url
+                .map(|url| url::Url::parse(&url))
+                .transpose()
+                .context("invalid KDC URL")?,
+            hostname: Some(destination),
+        }),
+    )
+    .await?;
```

Comment:
Why not pass the network client unconditionally?
```suggestion
    let mut network_client = WasmNetworkClient::new();

    let connection_result = ironrdp_futures::connect_finalize(
        upgraded,
        &mut framed,
        (&destination).into(),
        server_public_key,
        Some(&mut network_client),
        connector,
        Some(KerberosConfig {
            kdc_proxy_url: kdc_proxy_url
                .map(|url| url::Url::parse(&url))
                .transpose()
                .context("invalid KDC URL")?,
            hostname: Some(destination),
        }),
    )
    .await?;
```